src/common/core/corediff.js

```diff
@@ -2557,12 +2566,19 @@ define(['common/util/canon',
             // return done;
         };
 
+        /**
+         *
+         * @param {object} base - diff1
+         * @param {object} extension - diff2
+         *
+         * @returns {object}
+         */
         this.tryToConcatChanges = function (base, extension) {
             var result = {};
             _conflictItems = [];
             _conflictMine = {};
             _conflictTheirs = {};
-            _concatBase = base;
+            _concatBase = base; // FIXME: Are these meant to be copies, currently base and extension are mutated?
```

Comment:
I would say, that it is not an issue because of the current use-cases (the base input is never used after this, rather the result is the one that will be ultimately used), but it would be probably wiser to have the _concatBase as a deep copy and not mutate any input objects.